velox/functions/prestosql/tests/DecimalArithmeticTest.cpp

```diff
@@ -379,6 +379,55 @@ TEST_F(DecimalArithmeticTest, round) {
           DECIMAL(38, 1))});
 }
 
+TEST_F(DecimalArithmeticTest, roundN) {
+  // Round upto 'scale' decimal places.
+  testDecimalExpr<TypeKind::BIGINT>(
+      {makeShortDecimalFlatVector({123, 552, -999, 0}, DECIMAL(4, 3))},
+      "round(c0, CAST(3 as integer))",
+      {makeShortDecimalFlatVector({123, 552, -999, 0}, DECIMAL(3, 3))});
+  // Round upto scale-1 decimal places.
+  testDecimalExpr<TypeKind::BIGINT>(
+      {makeShortDecimalFlatVector({120, 550, -1000, 0}, DECIMAL(4, 3))},
+      "round(c0, CAST(2 as integer))",
+      {makeShortDecimalFlatVector({123, 552, -999, 0}, DECIMAL(3, 3))});
+  // Round upto 0 decimal places.
+  testDecimalExpr<TypeKind::BIGINT>(
+      {makeShortDecimalFlatVector({100, 600, -1000, 0}, DECIMAL(4, 2))},
+      "round(c0, CAST(0 as integer))",
+      {makeShortDecimalFlatVector({123, 552, -999, 0}, DECIMAL(3, 2))});
+  // Round upto -1 decimal places.
+  testDecimalExpr<TypeKind::BIGINT>(
+      {makeShortDecimalFlatVector({120, 550, -1000, 0}, DECIMAL(4, 1))},
+      "round(c0, CAST(0 as integer))",
+      {makeShortDecimalFlatVector({123, 552, -999, 0}, DECIMAL(3, 1))});
+  // Round upto -2 decimal places. Here precision == scale - decimals.
```

Comment:
Can you verify these results with Presto Java? We should add E2E tests as well after this is merged.

Reply:
I had verified while writing the tests. Will definitely add E2E tests.